src/subscriptions/SubscriptionsCard.js

```diff
@@ -23,13 +24,14 @@ export default class SubscriptionsCard extends PureComponent {
 
   handleFilterChange = (filter: string) => this.setState({ filter });
 
-  handleSwitchChange = (streamName: string, switchValue: boolean) => {
-    const { auth } = this.props;
+  handleSwitchChange = (streamName: string, switchValue: boolean, streamId: number) => {
+    const { auth, markStreamMessagesRead } = this.props;
 
     if (switchValue) {
       subscriptionAdd(auth, [{ name: streamName }]);
```

Comment:
We would likely need to handle the adding of unread messages when you subscribe again?